doc/manual/src/language/values.md

```diff
@@ -92,39 +92,48 @@
 
 - <a id="type-path" href="#type-path">Path</a>
 
-  *Paths*, e.g., `/bin/sh` or `./builder.sh`. A path must contain at
-  least one slash to be recognised as such. For instance, `builder.sh`
-  is not a path: it's parsed as an expression that selects the
-  attribute `sh` from the variable `builder`. If the file name is
-  relative, i.e., if it does not begin with a slash, it is made
-  absolute at parse time relative to the directory of the Nix
-  expression that contained it. For instance, if a Nix expression in
-  `/foo/bar/bla.nix` refers to `../xyzzy/fnord.nix`, the absolute path
-  is `/foo/xyzzy/fnord.nix`.
-
-  If the first component of a path is a `~`, it is interpreted as if
-  the rest of the path were relative to the user's home directory.
-  e.g. `~/foo` would be equivalent to `/home/edolstra/foo` for a user
-  whose home directory is `/home/edolstra`.
-
-  For instance, evaluating `"${./foo.txt}"` will cause `foo.txt` in the current directory to be copied into the Nix store and result in the string `"/nix/store/<hash>-foo.txt"`.
-
-  Note that the Nix language assumes that all input files will remain _unchanged_ while  evaluating a Nix expression.
+  *Paths* are distinct from strings and can be expressed by path literals such as `./builder.sh`.
+
+  Paths are the preferred type for referring to local files.
+  This is thanks to the following properties:
+  - Path values are always in a canonical form, so that you are relieved from trailing slashes, `.` and `..`.
```

Comment:
Unfortunately this isn't true, [paths can have a trailing slash](https://github.com/NixOS/nix/blob/538eb2617aab78f263dcfafff37784b9586ee7b9/src/libexpr/parser.y#L291-L294).

Reply:
But why is this then:

```shell
nix-repl> a = /home/                   
error: path has a trailing slash

       at «string»:1:2:

            1|  /home/
             |  ^
```

Reply:
```
nix-repl> let x = "x"; in /home${x}  
/homex

nix-repl> let x = "x"; in /home/${x}
/home/x
```

I.e. `/home/` is represented internally with a trailing slash, since canonicalizing it to `/home` would cause `/home${x}` to evaluate to `/homex`.

Reply:
This is internal to the `/p/${e}` expression and not observable as the result of a properly composable expression, right?
If it is, then I would expect that to be a bug.

At least these are fine:
```
nix-repl> /home${""}
/home

nix-repl> /home/${""}
/home

nix-repl> /home/${"/"}
/home

nix-repl> /home/${"////"}
/home
```

I should clarify that trailing slashes do matter in the literals.